File: 251004/Kalitsko/1lab/Rest/src/main/java/com/example/rest/mapper/PostMapper.java
package com.example.rest.mapper;

import com.example.rest.dto.PostRequestTo;
import com.example.rest.dto.PostResponseTo;
import com.example.rest.entity.Post;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;
import org.springframework.stereotype.Component;

@Mapper(componentModel = "spring")
public interface PostMapper {
    Post toEntity(PostRequestTo request);
    PostResponseTo toResponse(Post post);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/mapper/PostMapper.kt
package bsuir.dc.rest.mapper

import bsuir.dc.rest.dto.from.PostRequestTo
import bsuir.dc.rest.dto.to.PostResponseTo
import bsuir.dc.rest.entity.Issue
import bsuir.dc.rest.entity.Post

fun PostRequestTo.toEntity(issue: Issue): Post {
    return Post(
        id = this.id,
        issue = issue,
        content = this.content,
    )
}

fun Post.toResponse(): PostResponseTo {
    return PostResponseTo(
        id = this.id,
        issueId = this.issue.id,
        content = this.content
    )
}
